### PR TITLE
lib: check locale for alsabat

### PR DESCRIPTION
**Pull request by miRoox**
alsabat always returns -2 for "C.UTF-8" locale. Suggest user set locale to "C.utf8" to avoid this issue.

see https://github.com/alsa-project/alsa-utils/issues/192

Signed-off-by: Yong-an Lu <yongan.lu@intel.com>

### CODE REVIEW

File: case-lib/lib.sh [hunk 1]
```diff
@@ -694,7 +697,8 @@ is_firmware_file_zephyr()
 
     firmware_path=$(get_firmware_path)
     [ -n "$firmware_path" ] ||
-        die 'firmware path not found from journalctl, no firmware loaded or debug option disabled?'
+        dloge 'firmware path not found from journalctl, no firmware loaded or debug option disabled?'
+        return 2
```
Comment (marc-hb) [inline case-lib/lib.sh:701]: This looks like a git accident.

File: case-lib/lib.sh [hunk 2]
```diff
@@ -27,6 +27,9 @@ source "$SCRIPT_HOME/case-lib/hijack.sh"
     exit 2
 }
 
+# alsabat will return -2 for "C" locale
+export LC_ALL=en_US.UTF-8
```
Comment (marc-hb) [inline case-lib/lib.sh:31]: This change is good but wait: why is alsabat failing with a "C" locale and how? This sounds really bad. Please file a bug if none exists yet.

**Review by marc-hb — CHANGES_REQUESTED** [review]

**Review by marc-hb — APPROVED** [review]

**Review by marc-hb — CHANGES_REQUESTED** [review]
> alsabat returns -2 for "C" locale,

I tried to reproduce but the test passed:

```shell
export LC_ALL=C
ubuntu@jf-adlp-rvp-nocodec-8:~$ TPLG=/lib/firmware/intel/sof-tplg/sof-adl-nocodec.tplg MODEL=ADLP_RVP_NOCODEC_ZEPHYR ~/sof-test/test-case/check-alsabat.sh -p hw:sofnocodec,0 -c hw:sofnocodec,0 -C 2

PASS

```

We need more information.

File: case-lib/lib.sh (hunk 2 above)
Comment (marc-hb) [reply case-lib/lib.sh:31]: > Please file a bug if none exists yet.

We already have https://github.com/intel-innersource/drivers.audio.ci.sof-framework/issues/286 but it is: 1. private 2. incomplete.

**Comment by greg-intel** [comment]
SOFCI TEST

**Comment by miRoox** [comment]
The device test error is unrelated.

**Review by greg-intel — APPROVED** [review]

**Review by marc-hb — CHANGES_REQUESTED** [review]
Thanks @miRoox for filing https://github.com/alsa-project/alsa-utils/issues/192, very useful.

I did some testing and added more information in 192.

Based on my latest discoveries I think hardcoding the locale like this is too strict. Weird and very specific bug 192 aside, `alsabat` does not require any specific locale and neither does ALSA in general. So wee should not block people from testing with different locales. 

How about something like this instead in check-alsabat.sh:

```sh
if locale | grep -i 'C.utf-8'; then
  die 'Try C.utf8 instead, see https://github.com/alsa-project/alsa-utils/issues/192'
fi
```

I'll do it if you don't want to.

**Review by marc-hb — DISMISSED** [review]
Did you test this? I didn't.

**Comment by miRoox** [comment]
> Did you test this? I didn't.

Maybe died too early?

```
LANG=C.UTF-8
LC_CTYPE="C.UTF-8"
LC_NUMERIC=C.UTF-8
LC_TIME=C.UTF-8
LC_COLLATE="C.UTF-8"
LC_MONETARY=C.UTF-8
LC_MESSAGES="C.UTF-8"
LC_PAPER=C.UTF-8
LC_NAME=C.UTF-8
LC_ADDRESS=C.UTF-8
LC_TELEPHONE=C.UTF-8
LC_MEASUREMENT=C.UTF-8
LC_IDENTIFICATION=C.UTF-8
/home/ubuntu/sof-test/test-case/../case-lib/lib.sh: line 32: die: command not found
2023-02-23 05:17:55 UTC [ERROR] Starting func_exit_handler(), exit status=127, FUNCNAME stack:
2023-02-23 05:17:55 UTC [ERROR]  source()  @  /home/ubuntu/sof-test/test-case/../case-lib/lib.sh
2023-02-23 05:17:55 UTC [ERROR]  main()  @  /home/ubuntu/sof-test/test-case/verify-kernel-boot-log.sh:19
/home/ubuntu/sof-test/case-lib/hijack.sh: line 28: is_ipc4: command not found
/home/ubuntu/sof-test/case-lib/hijack.sh: line 104: is_ipc4: command not found
/home/ubuntu/sof-test/case-lib/hijack.sh: line 113: stop_test: command not found
2023-02-23 05:17:55 UTC [ERROR] Kernel check point "KERNEL_CHECKPOINT" is not properly set
2023-02-23 05:17:55 UTC [ERROR] KERNEL_CHECKPOINT=
2023-02-23 05:17:55 UTC [INFO] Test Result: FAIL!
```

**Comment by miRoox** [comment]
```
LANG=C.UTF-8
LC_CTYPE="C.UTF-8"
LC_NUMERIC=C.UTF-8
LC_TIME=C.UTF-8
LC_COLLATE="C.UTF-8"
LC_MONETARY=C.UTF-8
LC_MESSAGES="C.UTF-8"
LC_PAPER=C.UTF-8
LC_NAME=C.UTF-8
LC_ADDRESS=C.UTF-8
LC_TELEPHONE=C.UTF-8
LC_MEASUREMENT=C.UTF-8
LC_IDENTIFICATION=C.UTF-8
2023-02-23 11:40:48 UTC [ERROR] Try C.utf8 instead, see https://github.com/alsa-project/alsa-utils/issues/192
2023-02-23 11:40:48 UTC [ERROR] Starting func_exit_handler(), exit status=1, FUNCNAME stack:
2023-02-23 11:40:48 UTC [ERROR]  die()  @  /home/ubuntu/sof-test/test-case/../case-lib/lib.sh
2023-02-23 11:40:48 UTC [ERROR]  check_locale_for_alsabat()  @  /home/ubuntu/sof-test/test-case/../case-lib/lib.sh:790
2023-02-23 11:40:48 UTC [ERROR]  main()  @  /home/ubuntu/sof-test/test-case/check-alsabat.sh:61
2023-02-23 11:40:49 UTC [INFO] ktime=15629 sof-test PID=5829: ending
2023-02-23 11:40:49 UTC [INFO] Test Result: FAIL!
```

File: test-case/check-fw-echo-reference.sh [hunk 3]
```diff
@@ -20,8 +20,7 @@ set -e
 rm -f /tmp/bat.wav.*
 
 # shellcheck source=case-lib/lib.sh
-libdir=$(dirname "${BASH_SOURCE[0]}")
-source "$libdir"/../case-lib/lib.sh
+source "$(dirname "${BASH_SOURCE[0]}")"/../case-lib/lib.sh
```
Comment (marc-hb) [inline test-case/check-fw-echo-reference.sh:23]: I think you could have just moved the shellcheck comment one-line down instead but no big deal.

**Review by marc-hb — APPROVED** [review]

**Comment by marc-hb** [comment]
Usual suspend/resume failures in https://sof-ci.01.org/softestpr/PR993/build157/devicetest/index.html, everything else passing.

**Review by aiChaoSONG — APPROVED** [review]

**Review by keqiaozhang — APPROVED** [review]

**Comment by marc-hb** [comment]
The alsabat bug has just been fixed but let's keep that in place for at least a few months: sof-test should not require all users to upgrade alsa-utils immediately and removing one `-` character in the locale name should be a very small inconvenience.

**Pull request merged**